sweyer_plugin/lib/sweyer_plugin_method_channel.dart

```diff
@@ -7,41 +7,25 @@ import 'package:flutter/services.dart';
 import 'sweyer_plugin_platform_interface.dart';
 import 'sweyer_plugin.dart';
 
-class SweyerMethodChannelException extends SweyerPluginException {
-  const SweyerMethodChannelException._(this.value, [this._exception]);
-
-  final PlatformException? _exception;
-  @override
-  final String value;
-
+enum SweyerMethodChannelExceptionCode {
   /// Generic error.
-  static const unexpected = SweyerMethodChannelException._('UNEXPECTED_ERROR');
+  unexpected('UNEXPECTED_ERROR'),
 
   /// On Android 30 requests like `MediaStore.createDeletionRequest` require
   /// calling `startIntentSenderForResult`, which might throw this exception.
-  static const intentSender = SweyerMethodChannelException._('INTENT_SENDER_ERROR');
+  intentSender('INTENT_SENDER_ERROR'),
```

Comment:
where did the `intentSender` come?

Reply:
We were never specifically handling it, so it is just a generic `SweyerException` now. But it is captured in the `cause`.

Reply:
> We were never specifically handling it, so it is just a generic `SweyerException` now. But it is captured in the `cause`.

I think if we are doing this, all exception codes should have a separate exception for them

Reply:
The indent sender is very Android specific and not very descriptive, I decided to name the error `PlatformHandlerException` instead. Maybe I would need something like that on Windows as well. I hope that's ok for you.